src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/pre_compute.cpp

```diff
@@ -0,0 +1,182 @@
+    auto inv_freq = opp::wrap_type<ov::op::v0::Constant>();
+    auto inv_freq_convert = opp::optional<ov::op::v0::Convert>({inv_freq->output(0)});
+    auto broadcast = opp::wrap_type<ov::op::v3::Broadcast>({inv_freq_convert, concat_1});
+    auto position_ids = opp::wrap_type<ov::op::v0::Parameter>();
+    auto unsqueeze = opp::wrap_type<ov::op::v0::Unsqueeze>({position_ids, opp::wrap_type<ov::op::v0::Constant>()});
+    auto convert = opp::wrap_type<ov::op::v0::Convert>({unsqueeze});
+    auto matmul = opp::wrap_type<ov::op::v0::MatMul>({broadcast, convert});
+    auto transpose = opp::wrap_type<ov::op::v1::Transpose>({matmul, opp::any_input()});
+    auto concat_2 = opp::wrap_type<ov::op::v0::Concat>({transpose, opp::any_input()});
+    auto sin_cos = opp::wrap_type<ov::op::v0::Sin, ov::op::v0::Cos>({concat_2});
+
+    pattern = sin_cos;
+    verifier_cb = [=](ov::pass::pattern::Matcher& m) {
+        auto& node_to_output = m.get_pattern_value_map();
+
+        this->matched_position_ids = node_to_output.at(position_ids).get_node_shared_ptr();
+
+        this->matched_concat =  node_to_output.at(concat_1).get_node_shared_ptr();
+        auto matched_concat_2 = node_to_output.at(concat_2).get_node_shared_ptr();
+
+        // TODO: maybe check for it's dims
+        this->matched_inv_freq = node_to_output.at(inv_freq).get_node_shared_ptr();
+
+        // locating sin and cos
+        auto concat_users = matched_concat_2->output(0).get_target_inputs();
+        if (concat_users.size() != 2) {
+            LOG_INFO("Rope sin_cos invalid at: "<< matched_concat_2->get_name());
+            return false;
+        }
+        auto concat_it = concat_users.begin();
+        auto cos_node = concat_it->get_node();
+        concat_it++;
+        auto sin_node  = concat_it->get_node();
+
+        if (ov::as_type<ov::op::v0::Sin>(sin_node) == nullptr) {
+            std::swap(sin_node, cos_node);
+        }
+        if (ov::as_type<ov::op::v0::Sin>(sin_node) == nullptr ||
+            ov::as_type<ov::op::v0::Cos>(cos_node) == nullptr) {
+            LOG_INFO("Rope sin_cos nodes are not valid: sin=" << sin_node->get_name() << ", cos=" << cos_node->get_name());
+            return false;
+        }
+        LOG_INFO("Rope sin_cos nodes found : sin=" << sin_node->get_name() << ", cos=" << cos_node->get_name());
+        this->matched_cos = cos_node->shared_from_this();
+        this->matched_sin = sin_node->shared_from_this();
+
+        return true;
+    };
+}
+
+ov::npuw::patterns::pre_compute::RopeCacheMatcher::RopeCacheMatcher(const uint32_t max_prompt_len) {
+    auto rpe = std::make_shared<RopePatternLLama2>();
+
+    rpe->callback = [=](ov::pass::pattern::Matcher& m) -> bool {
+        auto inv_freq_size = ov::shape_size(rpe->matched_inv_freq->get_shape());
+
+        LOG_INFO("Making sin-cos cache of size: " << max_prompt_len << "x" << inv_freq_size);
+
```

Comment:
Please replace all `_INFO` with `_VERB` here (in the whole file). These logs are way too specific to be at `_INFO`.

Reply:
done